### PR TITLE
Feature/resolvers models

### PR DESCRIPTION
**Pull request by patrick-replogle**
# Description
I added a lot of the basic boilerplate for the resolvers. This includes model functions for events, users and categories, resolvers for those three categories, as well as some tests for each to get started. I cleaned up the file structure and split the resolvers into separate folders and files. Lastly, I adjusted the graphql schema to better match the new knex migration, so that the resolvers work better. Photos and modifiers still don't resolve, so we might have to write some custom scalars for those. Readme will still need to be updated for these changes.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?

- [x] I wrote tests for the changes, but they will likely need to be updated as we move forward and add things such as authentication. More basic unit tests will also likely be needed, since my tests are relying on the database at this time. 

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: src/models/users/user-models.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const db = require("../../../data/dbConfig.js");
+
+const newUser = {
+  Email: String(Math.random()),
+  Password: "secret",
+  FirstName: "John",
+  LastName: "Doe",
+  Address: "Portland",
+  Gender: "male",
+  Latitude: 22.222,
```
Comment (PaulMEdwards) [inline src/models/users/user-models.test.js:12]: Perhaps set this Latitude to negative?

File: src/models/users/user-models.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+const updatedUser = {
+  Email: String(Math.random()),
+  Password: "moreSecret",
+  FirstName: "Jane",
+  LastName: "Doe",
+  Address: "Portland",
+  Gender: "female",
+  Latitude: 22.222,
+  Longitude: 5.55,
```
Comment (PaulMEdwards) [inline src/models/users/user-models.test.js:24]: Perhaps set this Longitude to negative?

File: src/models/users/user-models.test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  beforeAll(async () => {
+    const users = await db("Users");
+    userCount = users.length;
+  });
+
+  test("creates a user", async () => {
+    const added = await userModel.add(newUser);
+    createdUserId = added.id;
+    const users = await db("Users");
+    expect(users).toHaveLength(userCount + 1);
```
Comment (PaulMEdwards) [inline src/models/users/user-models.test.js:40]: Again, this seems like a good approach, but are we certain the tests are executed sequentially?

File: src/resolvers/events/event-resolvers.test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    input: {
+      Date: "Dec 12, 2021",
+      Start_Time: "6:30",
+      Title: "Sushi and Sake Night",
+      user_id: 1,
+      Description: "Come to my place for tasy eats and drinks!",
+      category_id: 1,
+      Address: "Portland",
+      Latitude: 2.233,
+      Longitude: -2.23,
```
Comment (PaulMEdwards) [inline src/resolvers/events/event-resolvers.test.js:36]: Good work testing a negative value here!

File: src/resolvers/events/event-resolvers.test.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                id,
+                Title
+            }
+        }
+    `,
+  operationName: "addEvent",
+  variables: {
+    input: {
+      Date: "Dec 12, 2021",
+      Start_Time: "6:30",
```
Comment (PaulMEdwards) [inline src/resolvers/events/event-resolvers.test.js:29]: 6:30 am is kind of early for Sake...

File: src/resolvers/events/event-resolvers.test.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+          id: `${testId}`,
+          input: { Title: "Sushi and Japenese Beer Night" },
+        },
+      });
+    const parsed = JSON.parse(updated.text);
+    expect(updated.status).toBe(200);
+    expect(parsed.data.updateEvent.Title).toEqual(
+      "Sushi and Japenese Beer Night"
+    );
+  });
```
Comment (PaulMEdwards) [inline src/resolvers/events/event-resolvers.test.js:121]: Might want to add tests to confirm date/time values. Keep AM/PM times in mind.

File: src/models/users/user-models.test.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+  test("created user is attending zero events", async () => {
+    const attending = await eventModel.findEventsAttending(createdUserId);
+    expect(attending.length).toEqual(0);
+  });
+
+  test("created user is deleted", async () => {
+    const deleted = await userModel.remove(createdUserId);
+    const users = await userModel.find();
+    expect(users.length).toEqual(userCount);
```
Comment (PaulMEdwards) [inline src/models/users/user-models.test.js:72]: Should the expectation here be .toEqual(userCount - 1)?

File: src/resolvers/users/user-resolvers.test.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  operationName: "addUser",
+  variables: {
+    input: {
+      Email: String(Math.random()),
+      Password: "thisissecret",
+      FirstName: "John",
+      LastName: "Doe",
+      Gender: "male",
+      Latitude: 2.22,
+      Longitude: -2.22,
```
Comment (PaulMEdwards) [inline src/resolvers/users/user-resolvers.test.js:23]: Again, good that you're testing negative values.

File: knexfile.js [hunk 9]
```diff
@@ -1,52 +1,43 @@
...
-    // connection: {
-    //   host:     'localhost',
-    //   port:     5432,
-    //   user:     'neighborhoodchef',
-    //   password: 'J4mb4l4y4!',
-    //   database: 'neighborhoodchef_dev',
-    // },
-    connection: 'postgres://neighborhoodchef:J4mb4l4y4!@localhost:5432/neighborhoodchef_dev',
+    client: "pg",
+    connection: process.env.DATABASE_URL,
```
Comment (PaulMEdwards) [inline knexfile.js:4]: Do you need to include DATABASE_URL in the .env.example file? I don't see that change in this PR...

BTW, I've added the same connection string as POSTGRES_CONNECTION_STRING in my latest PR changes... If they're the same purpose, we should probably agree on which variable name to use. I'm not opposed to using DATABASE_URL.

File: data/dbConfig.js [hunk 10]
```diff
@@ -1,4 +1,4 @@
-const knex = require('knex');
-const config = require('../knexfile.js');
-const environment = process.env.NODE_ENV || 'development';
-module.exports = knex(config[environment]);
\ No newline at end of file
+const knex = require("knex");
+const config = require("../knexfile.js");
+const environment = process.env.DB_ENV || "development";
```
Comment (PaulMEdwards) [inline data/dbConfig.js:3]: Any particular reason to switch from NODE_ENV to DB_ENV?

**Comment by PaulMEdwards** [comment]
@patrick-replogle please update the PR description to remove unnecessary stuff, like "Fixes # (issue)", "Please delete options that are not relevant", etc. Please also complete the Checklist items you've performed.

File: data/dbConfig.js [hunk 11]
```diff
@@ -1,4 +1,4 @@
-const knex = require('knex');
-const config = require('../knexfile.js');
-const environment = process.env.NODE_ENV || 'development';
-module.exports = knex(config[environment]);
\ No newline at end of file
+const knex = require("knex");
```
[the remark this answers is not in the document]
Comment (patrick-replogle) [reply data/dbConfig.js:1]: I've been using prettier - code formatter and I also find it annoying that it does this. When I config single quotes to be true, it then changes all double quotes to single quotes. I would love a recommendation if you have one, but I'll mess with it more to see if I can prevent this behavior.

File: src/models/events/event-models.test.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+  Date: new Date(),
+  Start_Time: "6:00",
+  End_Time: "9:30",
+  Title: "BBQ Madness",
+  Description: "I like ribs, so come over and eat ribs",
+  user_id: 1,
+  category_id: 1,
+  Address: "My house",
+  Latitude: 12.222,
+  Longitude: 2.2333,
```
[the remark this answers is not in the document]
Comment (patrick-replogle) [reply src/models/events/event-models.test.js]: I've added negative coordinates to all of the test data examples and also cleaned up some of the code in general in the tests. Let me know if it looks better.

File: src/models/events/event-models.test.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+  beforeAll(async () => {
+    const events = await db("Events");
+    eventCount = events.length;
+  });
+
+  test("creates a new event", async () => {
+    const created = await eventModels.add(newEvent);
+    createdEventId = created.id;
+    const events = await db("Events").select();
+    expect(events.length).toEqual(eventCount + 1);
```
[the remark this answers is not in the document]
Comment (patrick-replogle) [reply src/models/events/event-models.test.js]: The jest docs say the tests run sequentially within each test file, but I got rid of a lot of the comparison length assertions to make things less brittle in general. I changed most of the insert tests to then search the database by id make sure the newly added row exists in the database instead.

File: knexfile.js (hunk 9 above)
Comment (patrick-replogle) [reply knexfile.js]: I reverted things back to NODE_ENV in dbConfig and POSTGRES_CONNECTION_STRING in the knexfile. I was originally having issues getting knex to connect to postgres when running tests, which is why changed things. It seems to work now though.

File: src/resolvers/events/event-resolvers.test.js (hunk 6 above)
Comment (patrick-replogle) [reply src/resolvers/events/event-resolvers.test.js:120]: I added AM/PM to make things more specific. I wasn't sure what the database would accept as a valid time, but it seems super flexible.

File: data/dbConfig.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (PaulMEdwards) [reply data/dbConfig.js]: Maybe try `js-beautify` which is `hookyqr.beautify` in the VS Code Extensions library.

**Pull request merged**